### ISSUE TITLE
Circuit is not opening on recordResultPredicate

### ISSUE THREAD

**Issue by RoopaRJ** [resilience4j/resilience4j#1835]
Resilience4j version:2.0.1

Java version:17

i am using spring boot version 3,
the issue i am facing is circuit is not opening on error codes example 404
below configuration i am using 
resilience4j:
  circuitbreaker:
    instances:
      abc:
        registerHealthIndicator: true
        slidingWindowType: TIME_BASED
        slidingWindowSize: 10
        failureRateThreshold: 50
        minimumNumberOfCalls: 1
        automaticTransitionFromOpenToHalfOpenEnabled: true
        permittedNumberOfCallsInHalfOpenState : 10
        slowCallDurationThreshold : 60s
        slowCallRateThreshold: 100
        writableStackTraceEnabled : true
        eventConsumerBufferSize: 10
        waitDurationInOpenState : 60s
        recordFailurePredicate: package.DefaultRetryableExceptionPredicate
        recordResultPredicate: package.DefaultRetryableResultPredicate

  retry:
    instances:
      abc:
        maxAttempts: 3
        waitDuration: 1s
        enableExponentialBackoff: false
        exponentialBackoffMultiplier: 1

**Comment by RobWin** [comment]
Hi,
is your HTTP client throwing an exception on HTTP status code 404?
Resilience4j is protocol agnostic and only catches exceptions.

**Comment by gopiprasanthpotipireddy** [comment]
Circuit Breaker is not transitioning to open state on  record result and for the same configuration record exception is working fine. we did the upgrade to  resilience4j  2.0.01 springboot 3  from resilience4j  1.7.0 springboot 2. 

this issue is after upgrade.

**Comment by RobWin** [comment]
@gavlyukovskiy Could be related to the last PR

**Comment by gavlyukovskiy** [comment]
@RobWin double checked the final configuration for `spring-boot2:2.0.0` and `spring-boot3:2.0.1`, both generate the same config (`circuitBreakerRegistry.circuitBreaker("abc").getCircuitBreakerConfig()`):
```
CircuitBreakerConfig {recordExceptionPredicate=package.DefaultRetryableExceptionPredicate@c6653e, ignoreExceptionPredicate=io.github.resilience4j.circuitbreaker.CircuitBreakerConfig$$Lambda$942/0x000000080131fa00@435e224d, recordExceptions=[], ignoreExceptions=[], failureRateThreshold=50.0, permittedNumberOfCallsInHalfOpenState=10, slidingWindowSize=10, slidingWindowType=TIME_BASED, minimumNumberOfCalls=1, writableStackTraceEnabled=true, automaticTransitionFromOpenToHalfOpenEnabled=true, waitIntervalFunctionInOpenState=io.github.resilience4j.core.IntervalFunction$$Lambda$946/0x000000080131dcc0@788fa91f, slowCallRateThreshold=100.0, slowCallDurationThreshold=PT1M}
```
which matches values from application properties. 
@gopiprasanthpotipireddy perhaps you have any `CircuitBreakerConfigCustomizer` defined? Otherwise small reproducible application would be very helpful

**Comment by gopiprasanthpotipireddy** [comment]
@gavlyukovskiy  not using any customizer ,using the below two.
recordFailurePredicate: package.DefaultRetryableExceptionPredicate
recordResultPredicate: package.DefaultRetryableResultPredicate

Added a reproducible here run the app and go to  -> host/result/failure

**expectation: circuit breaker should open 
actual: not opening circuit breaker**

same working fine for exceptions 
reproducible -> hostname:port/result/exception

[]( https://github.com/gopiprasanthpotipireddy/resilience4j-spring-boot3-demo
)

**Comment by gavlyukovskiy** [comment]
The only thing I could find is that in https://github.com/resilience4j/resilience4j/commit/0e4964c44bd7366ab2da3391ac72deeb8e3ae956 after any circuit breaker call that did not produce an exception it was calling `circuitBreaker.onResult` that checks `circuitBreakerConfig.getRecordResultPredicate().test(result)`, but it was changed to `circuitBreaker.onSuccess` that doesn't do that.
@RobWin what do you think?

**Comment by gavlyukovskiy** [comment]
Looks like a merge conflict handled bad, as the change was actually done in a2534c51 that adds `circuitBreaker .onSuccess`.

**Comment by gavlyukovskiy** [comment]
@RobWin I glanced over that commit and over all usages of `onSuccess` and I didn't find anything else, so opened a PR with a fix.

**Comment by RobWin** [comment]
Seems I need to improve some unit tests that this wasn' discovered when we removed Vavr.

**Comment by RobWin** [comment]
Indeed, here it happened:

![image](https://user-images.githubusercontent.com/5815046/206135363-a14989af-dc17-4827-a5d3-0294b3ff4016.png)

**Comment by RobWin** [comment]
@

**Comment by gopiprasanthpotipireddy** [comment]
verified @RobWin working fine now. @gavlyukovskiy  thanks for the PR

**Comment by gopiprasanthpotipireddy** [comment]
when is this one going to release ::)

**Comment by RobWin** [comment]
I won't wait too long. 
If there is no further bug within the next days, I could release a bugfix release soon.

**Comment by RobWin** [comment]
Released 2.0.2
